app/scripts/services/secrets.js

```diff
@@ -53,22 +53,25 @@ angular.module("openshiftConsole")
     };
 
     var decodeSecretData = function(secretData) {
-      var decodedSecretData = {};
-      _.each(secretData, function(encodedData, paramName) {
-        var decodedData = window.atob(encodedData);
+      return _.mapValues(secretData, function(data, paramName) {
+        var decodedData = window.atob(data);
```

Comment:
Maybe

```js
      return _.mapValues(secretData, function(data, paramName) {
        switch (paramName) {
          case ".dockercfg":
            return decodeDockercfg(window.atob(data));
          case ".dockerconfigjson":
            return decodeDockerconfigjson(window.atob(data));
          case "username":
          case "password":
          case ".gitconfig":
          case "ssh-privatekey":
          case "ca.crt":
            return window.atob(data);
          default:
            return data;
        }
      });
```

Might be better to call `window.atob()` inside `decodeDockercfg` and `decodeDockerconfigjson`. Decode is in the function name, which implies to me it will handle that for me.